pkg/registry/flowcontrol/rest/flowschema_test.go

```diff
@@ -0,0 +1,289 @@
+			current:   nil,
+			expected:  newFlowSchema("fs1", "pl1", 100).Object(),
+		},
+		{
+			name:      "suggested flow schema does not exist and we should not ensure - the object should not be created",
+			fsType:    "suggested",
+			strategy:  &suggestedFSEnsureStrategy{shouldCreate: false},
+			bootstrap: newFlowSchema("fs1", "pl1", 100).Object(),
+			current:   nil,
+			expected:  nil,
+		},
+		{
+			name:      "suggested flow schema exists, auto update is enabled, spec does not match - current object should be updated",
+			fsType:    "suggested",
+			strategy:  &suggestedFSEnsureStrategy{shouldCreate: true},
+			bootstrap: newFlowSchema("fs1", "pl1", 100).Object(),
+			current:   newFlowSchema("fs1", "pl1", 200).WithAutoUpdateAnnotation("true").Object(),
+			expected:  newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("true").Object(),
+		},
+		{
+			name:      "suggested flow schema exists, auto update is disabled, spec does not match - current object should not be updated",
+			fsType:    "suggested",
+			strategy:  &suggestedFSEnsureStrategy{shouldCreate: true},
+			bootstrap: newFlowSchema("fs1", "pl1", 100).Object(),
+			current:   newFlowSchema("fs1", "pl1", 200).WithAutoUpdateAnnotation("false").Object(),
+			expected:  newFlowSchema("fs1", "pl1", 200).WithAutoUpdateAnnotation("false").Object(),
+		},
+
+		// for mandatory configurations
+		{
+			name:      "mandatory flow schema does not exist - new object should be created",
+			fsType:    "mandatory",
+			strategy:  &mandatoryFSEnsureStrategy{},
+			bootstrap: newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("true").Object(),
+			current:   nil,
+			expected:  newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("true").Object(),
+		},
+		{
+			name:      "mandatory flow schema exists, annotation is missing - annotation should be added",
+			fsType:    "mandatory",
+			strategy:  &mandatoryFSEnsureStrategy{},
+			bootstrap: newFlowSchema("fs1", "pl1", 100).Object(),
+			current:   newFlowSchema("fs1", "pl1", 100).Object(),
+			expected:  newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("true").Object(),
+		},
+		{
+			name:      "mandatory flow schema exists, auto update is disabled, spec does not match - current object should be updated",
+			fsType:    "mandatory",
+			strategy:  &mandatoryFSEnsureStrategy{},
+			bootstrap: newFlowSchema("fs1", "pl1", 100).Object(),
+			current:   newFlowSchema("fs1", "pl1", 200).WithAutoUpdateAnnotation("false").Object(),
+			expected:  newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("true").Object(),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			clientset := fake.NewSimpleClientset()
+			if test.current != nil {
+				clientset.FlowcontrolV1beta1().FlowSchemas().Create(context.TODO(), test.current, metav1.CreateOptions{})
+			}
+
+			err := ensureFlowSchema(clientset.FlowcontrolV1beta1(), test.fsType, test.bootstrap, test.strategy)
+			if test.err != err {
```

Comment:
This would probably be too strict a test, if `test.err` were ever anything besides `nil`.
Since `test.err` is always `nil`, how about just writing `if err != nil {` here?

Reply:
yes, going with `if err != nil` now.